File: YoonJeong/011주차/같은숫자는싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList();
        
        for (int a : arr) {
            if (!answer.isEmpty() && answer.get(answer.size() - 1) == a) {
                continue;
            } else {
                answer.add(a);
            }
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: juhee2/11주차/같은숫자는싫어.py
# 효율성 0점
# def solution(arr):
#     remove = [i for i in range(len(arr) - 1) if arr[i] == arr[i+1]]
#     return [arr[i] for i in range(len(arr)) if i not in remove]

def solution(arr):
    answer = []
    for i,a in enumerate(arr):
        if i == 0:
            answer.append(a)
            continue
        if answer[-1] != a:
            answer.append(a)
    return answer
